### DISCUSSION TITLE
Subdomain per Organization

### DISCUSSION THREAD

**Discussion by TylerMills** [vercel/vercel#3950] [General] [answered]
Goals:
* Provide every organization their own subdomain. For example: `organization.mysite.com` 
* Allow organizations to customize their domain. For example `hello.organization.com`

Is the only way to accomplish this via SSR for every page that validates the subdomain, comparing to the one configured in the user's profile?

**Comment by flutterforsale** [comment]
Hey everybody,

I am also looking solution for my project. 
My project same https://github.com/zeit/static-fun

How i can using Export static sites feature or getStaticProps for per subdomain?

How can per customer re-deploy by his self on UI without Zeit.

Thanks!

**Reply by paulogdm** [reply]
Not possible yet. This would require heavy changes in how Now works. SSR/SPR is advised.

**Comment by paulogdm** [comment, marked as answer]
In order to serve content based on the subdomain, you need a dynamic code that decides what you should serve to whom. To be very clear:

- If you are using a wildcard domain, you must use serverless functions and caching strategies in order to make the sites fast. [SPR](https://zeit.co/blog/serverless-pre-rendering) is also advised.
- There is no way to generate N sites and distribute them across N subdomains. But the idea is interesting and we could release support for it if the demand justifies it.

Attacking the goals of the OP directly:

> Provide every organization their own subdomain. For example: organization.mysite.com

This is certainly possible on our platform. You need to use [wildcard domains](https://zeit.co/blog/wildcard-domains).

> Allow organizations to customize their domain. For example hello.organization.com

This is a little more difficult but perfectly doable. You need to:

1. Guide the user to add the right DNS records.
2. Add his domain to our platform.
3. Make the necessary adjustments on a project.

To do all of the above, you need to use our API. An Enterprise Agreement may be advised if you plan to launch a SaaS service.
